kong/conf_loader/init.lua

```diff
@@ -482,6 +482,10 @@ local function load(path, custom_conf, opts)
   -- load absolute paths
   conf.prefix = abspath(conf.prefix)
 
+  -- the runtime prefix is where we keep listening unix sockets for IPC and
+  -- private APIs
+  conf.runtime_prefix = pl_path.join(conf.prefix, "runtime")
```

Comment:
:wave: @fffonion says:

> I would actually suggest we just name it `sockets` and don't let user to configure it. Having it being named a broad name and make user facing may make usage of this directory out of our control in the future. From time to time we or user may decided to put some other stuff in it because it's called `runtime_prefix`, then break some setup again.

Quick clarification: it is not actually configurable by the user or exposed via `kong.conf` in any way. It's just a value derived from `prefix` that we're storing in the configuration object for convenience.

I chose the generic path+name of `runtime`+`runtime_prefix` to imply future intentions of relocating to a more "standard" mount point for runtime data (i.e. `/var/run/kong`) and potentially using the directory for other types of transient, runtime-scoped files.

That said, I'm totally okay with narrowing the scope and renaming to `sockets`/`socket_path` or something like that as long as a few of us can agree on it. We might also consider making it a "hidden" directory (`$prefix/runtime` => `$prefix/.runtime`) if that would help to inform operators that it's not something they should be touching themselves. @curiositycasualty, @Tieske thoughts?

Reply:
^ and @locao

Reply:
`/usr/local/kong/runtime/` indeed may mislead end users. Currently, `/usr/local/kong` is the actual runtime directory.

`/usr/local/kong/sockets/abc.sock` looks better, as this directory is only intended for internal sockets.

Reply:
> Quick clarification: it is not actually configurable by the user or exposed via kong.conf in any way. It's just a value derived from prefix that we're storing in the configuration object for convenience.

👍 In that case I would even suggest to put it in a constants (to ensure developer understand the readonly-ness of it).

Reply:
> I chose the generic path+name of runtime+runtime_prefix to imply future intentions of relocating to a more "standard" mount point for runtime data (i.e. /var/run/kong) and potentially using the directory for other types of transient, runtime-scoped files.

I agree to this 👍

Reply:
> put it in a constants (to ensure developer understand the readonly-ness of it)

I understand the concern, but this pattern of computing values within the conf loader and injecting them into `kong.configuration` is very common in our code already.

It cannot go into constants because it depends on the value of `prefix` in order to compute the absolute path (we've had [problems in the past](https://github.com/Kong/kong/pull/9337) where code dealing with unix sockets used a relative path instead of absolute).

Injecting this value into the global `kong.configuration` object also makes it readily available for use in our nginx.conf templates. This change is going to be backported across several versions, so using common, supported patterns is helpful in keeping the change size minimum and uncomplicated.

Reply:
I do agree doing the infer and put the new value back into kong.configuration is what we are doing
right now, it just seems we have no control of whether it's a "inferred" only value or "user configurable" value at all.
Let's keep it as is, but maybe add a comment in [here](https://github.com/Kong/kong/pull/13409/files#diff-25ec5b62312645a6a18cb1180612b544abb79a413db3f53211ea5d626fef095bR487) mentioning the intention of not exposing this value to user side, or simply attach the link to this PR discussion.
I can imagine the exact same issue would come bite us again if this is made configurable accidently.

It's common use case for other softwares to put stuff in a global or standard prefix for those data, again we need to consider the development pattern that uses container with mounts from different host FS, which makes a standard prefix not likely archievable in the future.